Framework/CurveFitting/src/Algorithms/PlotPeakByLogValue.cpp

```diff
@@ -190,6 +190,7 @@ void PlotPeakByLogValue::exec() {
   m_baseName = getPropertyValue("OutputWorkspace");
   std::vector<double> startX = getProperty("StartX");
   std::vector<double> endX = getProperty("EndX");
+  std::vector<std::vector<double>> exclude = getExclude(wsNames.size());
```

Comment:
Is the inner `std::vector<double>` holding a start and end for a masked x range? If so we could probably use a `std::pair` instead? 

Also would it be possible to do something like `using ExcludeRange = std::pair<double>;`, and then return a `std::vector<ExcludeRange>` to make it more explicit?

Reply:
This is a tricky one, Fit can accept multiple ranges of exclusion, so for example (-0.1, -0.05, 1, 2) will make two ranges (-0.1, -0.05) and (1, 2). the indirect interface does not allow for multiple ranges though. I'm not sure what the best solution for that is.

Reply:
This only appears to take one exclude range per vector<double> https://github.com/mantidproject/mantid/blob/ca5cd1343619cc18971a0c049a2b0219e8498914/Framework/CurveFitting/src/Algorithms/PlotPeakByLogValue.cpp#L550 , unless I am misunderstanding it?

I thought that the Mask X range next to `Fit Single Spectrum` accepts multiple exclude ranges?

Reply:
Yes I have notice that, I need to work out a way of telling `PlotPeaksByLogValue` that, right now if you put multuiple ranges in the entry in the table or lower box of the interface it will be used on the one after it. It may need some creative problem solving to fix

Reply:
Would it be something to fix in a separate PR, or in this PR?

Reply:
I am not sure at this point, I will have to see if there is a solution to this.